pkg/flow/internal/controller/component_node.go

```diff
@@ -0,0 +1,33 @@
+package controller
+
+import (
+	"context"
+
+	"github.com/grafana/agent/component"
+)
+
+// ComponentNode is a generic representation of a Flow component.
+type ComponentNode interface {
+	RunnableNode
+
+	// CurrentHealth returns the current health of the node.
+	CurrentHealth() component.Health
+
+	// Arguments returns the current arguments of the managed component.
+	Arguments() component.Arguments
+
+	// Exports returns the current set of exports from the managed component.
+	Exports() component.Exports
+
+	// Label returns the label for the block or "" if none was specified.
```

Comment:
All components have labels now, right? 

```suggestion
	// Label returns the component label.
```